src/main/java/data_access/StockDataAccessInterface.java

```diff
@@ -13,5 +14,6 @@ public interface StockDataAccessInterface {
      * @return a hashmap with the stock ticker as the key and the Stock entity as the value.
      * It should contain all stocks in the database.
      */
-    Map<String, Stock> getStocks() throws RateLimitExceededException;
+    // TODO: should this be a list instead of a map? as we are not using the key anywhere
+    Map<String, Stock> getStocks() throws RateLimitExceededException, IOException;
```

Comment:
I checked the use of `getStocks` in `MarketTracker` and indeed a list is sufficient here as we are updating all stocks. However, for extension concern, a map will be more efficient if we want to only update a specific stock or a given list of stocks. We can keep it as it is right now.